kernel/thread.c

```diff
@@ -237,6 +237,28 @@ static void _free_zombies(void* arg)
 
 void myst_zombify_thread(myst_thread_t* thread)
 {
+    // If the thread was blocked on a condvar and got killed,
+    // get out of the waiting list of the condvar.
+    // If the thread exits normally, the cond_wait should be NULL.
+    myst_cond_signal_thread(thread->signal.cond_wait, thread);
```

Comment:
curious: why not check `cond_wait != NULL` before calling? To me, EINVAL makes me feel it's an error.

Reply:
Works either way. I just prefer to keep the caller logic simpler.